### DISCUSSION TITLE
Instrumentation Skipped after Update

### DISCUSSION THREAD

**Discussion by codeforaliving** [open-telemetry/opentelemetry-java-instrumentation#6636] [Q&A]
We are trying to upgrade our Spring Boot app from 1.12.0 to 1.14.0 and noticed the following warnings in the app log:

> [main] WARN muzzleMatcher - Instrumentation skipped, mismatched references were found: spring-boot-actuator-autoconfigure [class io.opentelemetry.javaagent.instrumentation.spring.actuator.SpringBootActuatorInstrumentationModule] on jdk.internal.loader.ClassLoaders$AppClassLoader@9e89d68
> [main] WARN muzzleMatcher - -- io.opentelemetry.micrometer1shim.OpenTelemetryMeterRegistry:91 Missing method io.micrometer.core.instrument.distribution.HistogramGauges#registerWithCommonFormat(Lio/micrometer/core/instrument/LongTaskTimer;Lio/micrometer/core/instrument/MeterRegistry;)Lio/micrometer/core/instrument/distribution/HistogramGauges;
> [main] WARN muzzleMatcher - -- io.opentelemetry.javaagent.instrumentation.micrometer.v1_5.MicrometerSingletons:20 Missing method io.micrometer.core.instrument.MeterRegistry#newLongTaskTimer(Lio/micrometer/core/instrument/Meter$Id;)Lio/micrometer/core/instrument/LongTaskTimer;
> [main] WARN muzzleMatcher - -- io.opentelemetry.micrometer1shim.OpenTelemetryLongTaskTimer:36 Missing method io.micrometer.core.instrument.internal.DefaultLongTaskTimer#<init>(Lio/micrometer/core/instrument/Meter$Id;Lio/micrometer/core/instrument/Clock;Ljava/util/concurrent/TimeUnit;Lio/micrometer/core/instrument/distribution/DistributionStatisticConfig;Z)V

Does this have something to do with the Micrometer shim being moved out?  We are using Spring Boot Actuator 2.1.11.  We also noticed that no traces or metrics are exported when started.  Any ideas?  Are missing something in the upgrade?  Thanks in advance.

**Comment by mateuszrzeszutek** [comment]
Hey @codeforaliving ,
Spring Boot Actuator 2.1.11 is pretty old (~3 years), and it includes Micrometer version 1.1.9 -- our instrumentation targets versions 1.5+, and cannot support older versions. The log that you posted is basically saying that the Micrometer instrumentation had to be skipped because the Micrometer that's on the application classpath does not have all the methods that we're using inside our instrumentation.

**Reply by codeforaliving** [reply]
@mateuszrzeszutek Thanks for the reply.  Are you aware of a way to get this working without having to update Spring Boot Actuator?

**Reply by mateuszrzeszutek** [reply]
Unfortunately not. If you need the micrometer metrics bridge you have to update your dependencies.

**Reply by codeforaliving** [reply]
@mateuszrzeszutek ok, thanks for the response.  Unfortunately updating Spring Boot Actuator is a big deal for us right now.  Do you know if Spring dependency override versions will work in this case?  Anyone else run into this issue?

**Reply by mateuszrzeszutek** [reply]
Hmm, I guess you could try updating just the Micrometer version -- although you might run into runtime compatibility issues with Actuator; in case it uses one of the methods that no longer exist in the newer Micrometer versions.

**Reply by codeforaliving** [reply]
@mateuszrzeszutek for example, if I define `project.ext['micrometer.version'] = 'x.y.z'` in our project gradle file, shouldn't that work?  Do you know the exact version of micrometer that v1.14.0 of the agent is looking for?

**Reply by mateuszrzeszutek** [reply]
The minimum is 1.5.0, we're supporting everything >= 1.5.0.
You'll have to test it out, I've no idea if Actuator 2.1.11 will be able to work with Micrometer 1.5+
